File: day_1/c/distance.c
#include <stdio.h>
#include <stdlib.h>

int comp(const void* a, const void* b) {
    return (*(int*)a - *(int*)b);
}

void main() {
  int distance = 0;

  int arr[2000];
  for (int i = 0; i < 2000; i++) {
    scanf("%d", &arr[i]);
  }

  int n = 1000;

  int left[n];
  int right[n];

  for (int i = 0; i < n; i++) {
    left[i] = arr[i * 2];
    right[i] = arr[(i * 2) + 1];
  }
  qsort(left, n, sizeof(int), comp);
  qsort(right, n, sizeof(int), comp);

  for (int i = 0; i < n; i++) {
    distance += abs(left[i] - right[i]);
  }
  printf("%i\n", distance);
}

File: day_1/go/distance.go
package main

import (
	"fmt"
	"slices"
)

func main() {
	a := []int{}
	for i := 0; i < 2000; i++ {
		var n int
		fmt.Scan(&n)
		a = append(a, n)
	}
	left := []int{}
	right := []int{}

	for i := 0; i < 1000; i++ {
		left = append(left, a[2 * i])
		right = append(right, a[1 + (2 * i)])
	}
	slices.Sort(left)
	slices.Sort(right)

	distance := 0
	
	for i := 0; i < 1000; i++ {
		distance += intAbs(left[i], right[i])
	}
	fmt.Println(distance)
}

func intAbs(x, y int) int {
	if x < y {
		return y - x
	}
	return x - y
}
